perception/autoware_tensorrt_bevdet/src/ros_utils.cpp

```diff
@@ -0,0 +1,137 @@
+  const std::vector<std::string> & class_names, float score_thre, const bool has_twist = true)
+{
+  for (auto b : boxes) {
+    if (b.score < score_thre) continue;
+    autoware_perception_msgs::msg::DetectedObject obj;
+
+    Eigen::Vector3f center(b.x, b.y, b.z + b.h / 2.);
+
+    obj.existence_probability = b.score;
+    // classification
+    autoware_perception_msgs::msg::ObjectClassification classification;
+    classification.probability = 1.0f;
+    if (b.label >= 0 && static_cast<size_t>(b.label) < class_names.size()) {
+      classification.label = getSemanticType(class_names[b.label]);
+    } else {
+      classification.label = Label::UNKNOWN;
+    }
+    obj.classification.emplace_back(classification);
+
+    // pose and shape
+    geometry_msgs::msg::Point p;
+    p.x = center.x();
+    p.y = center.y();
+    p.z = center.z();
+    obj.kinematics.pose_with_covariance.pose.position = p;
+
+    tf2::Quaternion q;
+    q.setRPY(0, 0, b.r);
+    obj.kinematics.pose_with_covariance.pose.orientation = tf2::toMsg(q);
+
+    obj.shape.type = autoware_perception_msgs::msg::Shape::BOUNDING_BOX;
+
+    geometry_msgs::msg::Vector3 v;
+    v.x = b.l;
+    v.y = b.w;
+    v.z = b.h;
+    obj.shape.dimensions = v;
+    if (has_twist) {
+      float vel_x = b.vx;
+      float vel_y = b.vy;
+      geometry_msgs::msg::Twist twist;
+      twist.linear.x = std::sqrt(std::pow(vel_x, 2) + std::pow(vel_y, 2));
+      twist.angular.z = 2 * (std::atan2(vel_y, vel_x) - b.r);
+      obj.kinematics.twist_with_covariance.twist = twist;
+      obj.kinematics.has_twist = has_twist;
+    }
+
+    bevdet_objects.objects.emplace_back(obj);
+  }
+}
+
+void getTransform(
+  const geometry_msgs::msg::TransformStamped & transform, Eigen::Quaternion<float> & rot,
+  Eigen::Translation3f & translation)
+{
+  rot = Eigen::Quaternion<float>(
+    transform.transform.rotation.w, transform.transform.rotation.x, transform.transform.rotation.y,
+    transform.transform.rotation.z);
+  translation = Eigen::Translation3f(
+    transform.transform.translation.x, transform.transform.translation.y,
+    transform.transform.translation.z);
+}
+
+void getCameraIntrinsics(
+  const sensor_msgs::msg::CameraInfo::SharedPtr msg, Eigen::Matrix3f & intrinsics)
+{
+  intrinsics << msg->k[0], msg->k[1], msg->k[2], msg->k[3], msg->k[4], msg->k[5], msg->k[6],
+    msg->k[7], msg->k[8];
+}
+
+void imageTransport(std::vector<cv::Mat> imgs, uchar * out_imgs, size_t width, size_t height)
+{
+  uchar * temp = new uchar[width * height * 3];
+  uchar * temp_gpu = nullptr;
+  CHECK_CUDA(cudaMalloc(&temp_gpu, width * height * 3));
+
+  for (size_t i = 0; i < imgs.size(); i++) {
+    cv::cvtColor(imgs[i], imgs[i], cv::COLOR_BGR2RGB);
+    CHECK_CUDA(cudaMemcpy(temp_gpu, imgs[i].data, width * height * 3, cudaMemcpyHostToDevice));
+    convert_RGBHWC_to_BGRCHW(temp_gpu, out_imgs + i * width * height * 3, 3, height, width);
+  }
+  delete[] temp;
+  CHECK_CUDA(cudaFree(temp_gpu));
+}
```

Comment:
@Shin-kyoto AutoCore has moved out the copied code as the vendor package from the suggestions you made here: https://github.com/autowarefoundation/autoware.universe/pull/7956#discussion_r1776330267.

Now **most** of the code is following the coding guideline I think except for this code that actually converts the image format into a input format that is expected by the library.

Do you think this would be acceptable?